app/uk/gov/hmrc/tradergoodsprofilesdatastore/connectors/RouterConnector.scala

```diff
@@ -57,18 +59,42 @@ class RouterConnector @Inject() (config: Configuration, httpClient: HttpClientV2
       .setHeader(clientIdHeader)
       .withBody(Json.toJson(traderProfile))
       .execute[HttpResponse]
-      .map(_ => Done)
+      .map { response =>
+        response.status match {
+          case OK => Done
+        }
+      }
 
   def getRecords(
     eori: String,
     lastUpdatedDate: Option[String] = None,
     page: Option[Int] = None,
     size: Option[Int] = None
-  )(implicit hc: HeaderCarrier): Future[HttpResponse] =
+  )(implicit hc: HeaderCarrier): Future[GetRecordsResponse] = {
+    var queryParams: Map[String, String] = Map.empty
```

Comment:
var? :O

Reply:
it's edited depending on which query params come in - coudn't think of a better way but there is probably some functional magic

Reply:
This seems to work. Passes the unit tests at least

    val lastUpdatedDatePair = lastUpdatedDate.map(date => ("lastUpdatedDate", date))
   val pagePair = page.map(page => ("page", page.toString))
    val sizePair = size.map(size => ("size", size.toString))

    val queryParams: Map[String, String] = List(lastUpdatedDatePair, pagePair, sizePair)
      .foldLeft(Map.empty[String, String])( (cur, pair) => cur ++ pair)